openedx/core/djangoapps/content_libraries/api.py

```diff
@@ -1001,12 +1002,15 @@ def get_bundle_links(library_key):
             opaque_key = libraries_linked[link_data.bundle_uuid].library_key
         except KeyError:
             opaque_key = None
-        # Append the link information:
+        try:
+            latest_version = blockstore_cache.get_bundle_version_number(link_data.bundle_uuid)
+        except BundleNotFound:
+            latest_version = 0
```

Comment:
@swalladge When I run this change and lx master, I get an error when editing the question set, but it seems to be only on the LX side.

If I revert this code to `continue` instead (like you did in a previous change), there's no error.

@bradenmacdonald is it ok to log the `BundleNotFound` error and skip adding the link, since it's an invalid link?

LX exception reported in JS console:

![400](https://user-images.githubusercontent.com/7556571/131973866-cf3c55a4-9e8e-4a81-b02e-f14f1a16acaf.png)

LX error logs:
```
2021-09-03 08:08:57,804 INFO 237 [labxchange.apps.library.api.v1.block_editing] block_editing.py:239 - CLONE_FOR_TRANSLATE current_data: {'display_name': 'Answer me these questions three', 'child_blocks': [{'usage_id': 'lb:LabXchange:f5ddc266:problem:1a0dc2012919a01fdf3e', 'block_type': 'problem', 'display_name': 'Question one', 'graded': True, 'max_attempts': None, 'weight': 1}, {'usage_id': 'lb:LabXchange:f5ddc266:problem:2e9eb30ff6fef53756e2', 'block_type': 'problem', 'display_name': 'Question two', 'graded': True, 'max_attempts': None, 'weight': 1}, {'usage_id': 'lb:LabXchange:f5ddc266:problem:fb0230101410e05fad86', 'block_type': 'problem', 'display_name': 'Question three', 'graded': True, 'max_attempts': None, 'weight': 1}]}
2021-09-03 08:08:57,913 INFO 237 [labxchange.apps.library.api.v1.block_editing] block_editing.py:243 - CLONE_FOR_TRANSLATE old_olx_str: <lx_assignment display_name="Answer me these questions three">
  <xblock-include definition="problem/1" source="1a0dc2012919a01fdf3e" usage="1a0dc2012919a01fdf3e"/>
  <xblock-include definition="problem/1" source="2e9eb30ff6fef53756e2" usage="2e9eb30ff6fef53756e2"/>
  <xblock-include definition="problem/1" source="fb0230101410e05fad86" usage="fb0230101410e05fad86"/>
</lx_assignment>
2021-09-03 08:08:58,129 WARNING 237 [django.request] log.py:228 - Bad Request: /api/v1/items/lb:LabXchange:f5ddc266:lx_assignment:1/blocks/lb:LabXchange:f5ddc266:lx_assignment:1/edit/
2021-09-03 08:08:58,129 WARNING 237 [django.request] log.py:228 - Bad Request: /api/v1/items/lb:LabXchange:f5ddc266:lx_assignment:1/blocks/lb:LabXchange:f5ddc266:lx_assignment:1/edit/
2021-09-03 08:08:58,130 WARNING 237 [django.server] basehttp.py:154 - "POST /api/v1/items/lb%3ALabXchange%3Af5ddc266%3Alx_assignment%3A1/blocks/lb%3ALabXchange%3Af5ddc266%3Alx_assignment%3A1/edit/ HTTP/1.1" 400 46
2021-09-03 08:08:58,326 INFO 237 [elasticsearch] base.py:170 - PUT http://labxchange.devstack.elasticsearch:9200/items/doc/lb%3ALabXchange%3Af5ddc266%3Alx_assignment%3A1 [status:200 request:0.008s]
2021-09-03 08:08:58,338 INFO 237 [django.server] basehttp.py:154 - "DELETE /api/v1/items/lb%3ALabXchange%3Af5ddc266%3Alx_assignment%3A1/image HTTP/1.1" 200 20
2021-09-03 08:08:58,470 WARNING 237 [labxchange.apps.library.api.v1.serializers] serializers.py:428 - Warning: Updating of item field item_type has not been implemented yet.
2021-09-03 08:08:58,471 WARNING 237 [labxchange.apps.library.api.v1.serializers] serializers.py:428 - Warning: Updating of item field is_block_change has not been implemented yet.
2021-09-03 08:08:58,505 INFO 237 [elasticsearch] base.py:170 - PUT http://labxchange.devstack.elasticsearch:9200/items/doc/lb%3ALabXchange%3Af5ddc266%3Alx_assignment%3A1 [status:200 request:0.007s]
2021-09-03 08:08:58,519 INFO 237 [celery.app.trace] trace.py:124 - Task labxchange.apps.library.api.v1.views.update_replicas_of[a84dbe83-8838-43ee-a088-4d07c88e7906] succeeded in 0.004064782056957483s: None
2021-09-03 08:08:58,545 INFO 237 [django.server] basehttp.py:154 - "PATCH /api/v1/items/lb%3ALabXchange%3Af5ddc266%3Alx_assignment%3A1 HTTP/1.1" 200 867
2021-09-03 08:08:58,753 INFO 237 [django.server] basehttp.py:154 - "GET /api/v1/items/lb%3ALabXchange%3Af5ddc266%3Alx_assignment%3A1 HTTP/1.1" 200 867
2021-09-03 08:08:58,796 INFO 237 [django.server] basehttp.py:154 - "GET /api/v1/xblocks/lb%3ALabXchange%3Af5ddc266%3Alx_assignment%3A1/handler_url HTTP/1.1" 200 146
2021-09-03 08:08:58,861 INFO 237 [django.server] basehttp.py:154 - "GET /api/v1/xblocks/lb%3ALabXchange%3Af5ddc266%3Alx_assignment%3A1/student_view_data HTTP/1.1" 200 602
2021-09-03 08:08:58,992 INFO 237 [django.server] basehttp.py:154 - "GET /api/v1/xblocks/lb%3ALabXchange%3Af5ddc266%3Alx_assignment%3A1/view/student_view HTTP/1.1" 200 21025
2021-09-03 08:08:59,170 INFO 237 [django.server] basehttp.py:154 - "GET /api/v1/items/lb%3ALabXchange%3Af5ddc266%3Alx_assignment%3A1/languages HTTP/1.1" 200 87
2021-09-03 08:08:59,244 INFO 237 [django.server] basehttp.py:154 - "GET /api/v1/xblocks/lb%3ALabXchange%3Af5ddc266%3Alx_assignment%3A1/student_view_user_state HTTP/1.1" 200 345
```

Reply:
@pomegranited You seem to be missing @swalladge 's [recent change](https://gitlab.com/opencraft/client/LabXchange/labxchange-dev/-/commit/375804d2516ffff9099ae26b45671999d39270b9) to [this line](https://gitlab.com/opencraft/client/LabXchange/labxchange-dev/-/blob/6996964a176982dba3de835b9fb5d74555522f22/backend/labxchange/service_clients/edxapp/serializers.py#L36) which sets `allow_null=True` for that `opaque_key` field that's giving the error.

> is it ok to log the BundleNotFound error and skip adding the link, since it's an invalid link?

It's not an invalid link though. If I understand correctly, the link works fine. It's just that the information about how to update the link to the latest version of the content (in this case, the latest version of the assessment) is NULL because the target content (the assessment) has been deleted. But the current version will continue to work (I think).

If you look at the code, [LabXchange doesn't even look at the result of `get_library_link`](https://gitlab.com/opencraft/client/LabXchange/labxchange-dev/-/blob/6996964a176982dba3de835b9fb5d74555522f22/backend/labxchange/apps/library/api/v1/block_editing.py#L466) other than to check if the link exists or not. So it shouldn't really matter if some fields are now NULL that weren't before, except that the serializer isn't expecting those to be null. So we just have to fix the serializer, which should be very simple to do, and which I think @swalladge has already done.

Reply:
Ok, thanks @bradenmacdonald .  Yes, can confirm that the serializer has been updated, so @pomegranited if you pull latest master and retest, it should work for you. :)

Reply:
@bradenmacdonald  @swalladge  My mistake! I thought I was running latest lx. But I've just done another pull and this is working exactly as expected. Thank you!

Reply:
Awesome!